### PR TITLE
desktop access: show an error when session is locked or times out

### LINKED ISSUE #10595: Make locked desktop connection attempts show error
### Description

**What happened**:
If a user tries to connect to a desktop that's been locked, the UI hangs indefinitely on the processing spinner

**What you expected to happen**:
An error message should be shown

### Reproduction Steps
1. [Place a lock a desktop](https://goteleport.com/docs/access-controls/guides/locking/#step-12-create-a-lock)
2. Attempt to connect to it

### PR DESCRIPTION
**Pull request by ibeckermayer**
fixes https://github.com/gravitational/teleport/issues/10595

### CODE REVIEW

File: lib/srv/desktop/windows_server.go [hunk 1]
```diff
@@ -754,14 +754,15 @@ func (s *WindowsService) handleConnection(proxyConn *tls.Conn) {
...
+	if err := s.connectRDP(ctx, tdpConn, desktop, authContext); err != nil {
 		log.Errorf("RDP connection failed: %v", err)
 		sendTDPError("RDP connection failed.")
 		return
 	}
 }
 
-func (s *WindowsService) connectRDP(ctx context.Context, log logrus.FieldLogger, tdpConn *tdp.Conn, desktop types.WindowsDesktop, authCtx *auth.Context) error {
+func (s *WindowsService) connectRDP(ctx context.Context, tdpConn *tdp.Conn, desktop types.WindowsDesktop, authCtx *auth.Context) error {
+	log := s.cfg.Log
```
Comment (jakule) [inline lib/srv/desktop/windows_server.go:765]: This looks a bit hacky as you're shadowing the package name. The implementation below also uses `log` or `s.cfg.Log`. Can we agree on one alias for the logger and either use `s.cfg.Log` or rename log to indicate that it's not the global logger? You could also embed logger into `WindowsService` to use `s.Infof(...)`.

**Review by jakule — APPROVED** [review]

File: lib/srv/desktop/windows_server.go [hunk 2]
```diff
@@ -1401,3 +1406,20 @@ type otherName struct {
...
+	log     logrus.FieldLogger
+	tdpConn *tdp.Conn
+}
+
+func (m *monitorErrorSender) WriteString(s string) (n int, err error) {
+	if err := m.tdpConn.SendError(s); err != nil {
+		m.log.Errorf("Failed to send TDP error message %v", err)
+	}
+
+	return 0, nil
```
Comment (zmb3) [inline lib/srv/desktop/windows_server.go:1424]: You're supposed to return the number of bytes that were written. Let's fix this hard-coded 0.

File: lib/srv/desktop/windows_server.go [hunk 3]
```diff
@@ -865,6 +866,10 @@ func (s *WindowsService) connectRDP(ctx context.Context, log logrus.FieldLogger,
 		Tracker:           rdpc,
 		TeleportUser:      identity.Username,
 		ServerID:          s.cfg.Heartbeat.HostUUID,
+		MessageWriter: &monitorErrorSender{
```
Comment (zmb3) [inline lib/srv/desktop/windows_server.go:869]: This doesn't look properly formatted.

File: lib/srv/desktop/windows_server.go (hunk 1 above)
Comment (ibeckermayer) [reply lib/srv/desktop/windows_server.go]: I thought I'd found a place where we were unnecessarily passing the logger through functions when we had it right on the `*WindowsService`, but now that I'm looking more closely I realize that we're actually adding fields to the logger [here](https://github.com/gravitational/teleport/blob/27c57f9a76a813fa943465a8d4e9c2bc17e12d4e/lib/srv/desktop/windows_server.go#L710) and [here](https://github.com/gravitational/teleport/blob/27c57f9a76a813fa943465a8d4e9c2bc17e12d4e/lib/srv/desktop/windows_server.go#L736) and [here](https://github.com/gravitational/teleport/blob/27c57f9a76a813fa943465a8d4e9c2bc17e12d4e/lib/srv/desktop/windows_server.go#L753), which is why we were passing it through. Basically `s.cfg.Log` is the global service logger, and then we derive a connection-specific log from it per connection. I've changed the logic back to reflect that in 6f3e479cdc127e1ab75eb667539b403c9e0afd59 

> This looks a bit hacky as you're shadowing the package name.

Unless I'm misunderstanding you, I don't think that's actually the case here. The package name is `logrus` and we're calling this thing `log`.

File: lib/srv/desktop/windows_server.go (hunk 3 above)
Comment (ibeckermayer) [reply lib/srv/desktop/windows_server.go:868]: I agree, but that's what it gives me 🤷‍♂️

File: lib/srv/desktop/windows_server.go (hunk 2 above)
Comment (ibeckermayer) [reply lib/srv/desktop/windows_server.go]: e5c885e4ac2fe13d39a809aebfef9c4a5034e0f8

File: lib/srv/desktop/windows_server.go (hunk 3 above)
Comment (zmb3) [reply lib/srv/desktop/windows_server.go:868]: huh. same here 🤷‍♂️

**Review by zmb3 — APPROVED** [review]

**Pull request merged**